ca/server.go

```diff
@@ -740,11 +737,13 @@ func (s *Server) signNodeCert(ctx context.Context, node *api.Node) error {
 	)
 
 	// Try using the external CA first.
-	cert, err := externalCA.Sign(ctx, PrepareCSR(rawCSR, cn, ou, org))
+	s.secConfigMu.Lock()
```

Comment:
I'm not sure about this.

It means a long-running or hanging `Sign` operation can block the security config update indefinitely.

It also means that multiple node certs can not be signed concurrently (though I guess the current code doesn't ever attempt this).